syntax = "proto3";

package person;

service PersonService {
    rpc Create (PersonMessage) returns (PersonMessage) {}
    rpc Retrieve (PersonId) returns (PersonMessage) {}
    rpc RetrieveAll (Empty) returns (PersonList) {}
}

message Empty {}

message PersonId {
    int32 id = 1;
}

message PersonMessage {
    int32 id = 1;
    string first_name = 2;
    string last_name = 3;
    string company_name = 4;
}

message PersonList {
    repeated PersonMessage persons = 1;
}
